Dockerng unit tests fixes: isolate global variables

rallytime opened the pull request:
### What does this PR do?
This fixes 10 of the 16 failing tests currently happening in the `unit.modules.dockerng_test` file. The `unit.states.dockerng_test` file set some global variables that were adversely affecting the module test run and causing the dockerng tests to fail. 

This isolates some of the versioning information/mocking so that the dockerng state and module unit test files won't step on each others toes by isolating some of these global variables using mocks.

There are still 6 test failures remaining, but I can't isolate the problem for the remaining 6 tests yet.

I will still keep trying to find the problem with these remaining 6 tests, but I wanted to get this in for now since the changes seem to be related to different global setting issues. (This one was with the docker version calls, the other is related to the `_refresh_mine_cache` decorator.)

### Previous Behavior
16 test failures in the `unit.modules.dockerng_test` file.

### New Behavior
10 tests are now passing, 6 are still failing for a different, unknown reason. 

### Tests written?

No - fixes existing tests.

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

Commit:
Clean up dockerng unit tests to avoid global variables and fixup some patching

Commit:
Add a couple more patches for docker.version information

salt-jenkins commented:
@rallytime, thanks for your PR! By analyzing the history of the files in this pull request, we identified @ticosax, @terminalmage and @dmacvicar to be potential reviewers.

Commit:
Add one more dockerng.version mock that was missed previously

Commit:
Mock out the get_client_args mocks in the dockerng module tests more aggressively

The tests pass fine when run independently, but when the full unit test suite
runs, the dockerng state tests are setting global values. This change makes the
module tests avoid those problem by relying on its own mocks.

rallytime commented:
The last commit should fix the remaining 6 dockerng test failures. 🎉 Thanks to @terminalmage for the help tracking those down.
